internal/email/email.go

```diff
@@ -61,56 +108,84 @@ func NewMessage(
 	return message.NewMessage(id.String(), payload), nil
 }
 
-// getBodyHTML returns the HTML body for the email based on the message payload
-func getEmailBodyHTML(ctx context.Context, inviteURL, minderURL, sponsor, project, role, inviteeEmail string) string {
-	data := struct {
-		AdminName        string
-		OrganizationName string
-		InvitationURL    string
-		RecipientEmail   string
-		MinderURL        string
-		TermsURL         string
-		PrivacyURL       string
-		SignInURL        string
-		RoleName         string
-		RoleVerb         string
-	}{
-		AdminName:        sponsor,
-		OrganizationName: project,
-		InvitationURL:    inviteURL,
-		RecipientEmail:   inviteeEmail,
-		MinderURL:        minderURL,
-		TermsURL:         DefaultMinderTermsURL,
-		PrivacyURL:       DefaultMinderPrivacyURL,
-		SignInURL:        minderURL,
-		RoleName:         role,
-		RoleVerb:         authz.AllRolesVerbs[authz.Role(role)],
-	}
-
-	// TODO: Load the email template from elsewhere
+// getHTMLBodyString returns the HTML body for the email based on the message payload.
+// If there is an error creating the HTML body, it will try to return the text body instead
+func getEmailBodyHTML(ctx context.Context, data bodyData) (string, error) {
+	var b strings.Builder
+	bHTML := bodyHTML
 
-	// Parse the template
-	tmpl, err := templates.Parse(bodyHTML)
+	bodyHTMLTmpl, err := util.NewSafeHTMLTemplate(&bHTML, "body-invite-html")
 	if err != nil {
-		zerolog.Ctx(ctx).Error().Err(err).Msg("error parsing the HTML template for email invitations")
-		// Default to the text body
-		return getEmailBodyText(inviteURL, sponsor, project, role)
+		zerolog.Ctx(ctx).Error().Err(err).Msg("error creating the HTML template for email invitations")
+		return getEmailBodyText(ctx, data)
```

Comment:
Is it safe to return a text body template if the HTML template fails to load?  I suspect not, as there are characters that are safe in text that are unsafe in HTML.